src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -498,14 +496,7 @@ private static T UnwrapAggregateException<T>(Func<T> action)
 
         private static DataServiceTransportException ConvertToDataServiceTransportException(WebException webException)
         {
-            HttpWebResponseMessage errorResponseMessage = null;
-            if (webException.Response != null)
-            {
-                HttpWebResponse httpResponse = (HttpWebResponse)webException.Response;
-                errorResponseMessage = new HttpWebResponseMessage(httpResponse);
-            }
-
-            return new DataServiceTransportException(errorResponseMessage, webException);
+            return new DataServiceTransportException(response: null, webException);

```

Comment:
Curious how the thrown exception looks like when we pass a null here. We should confirm. Maybe we should add a resource string with an appropriate error message to use here.

Reply:
I just noticed that the first parameter is of type `IODataResponseMessage`. Is there anything we can pass in place of what was being passed before? The question about the exception appears to the caller when a null is passed is still an important one to answer

Reply:
The code before was passing null if `webException.Response` was null. But `webException.Response` was non-null only when using `HttpWebResponse`. I didn't find a way to reliably get the `HttpResponseMessage` from an exception. From example, with an `OperationCancelledException`, it doesn't have a link to the `HttpResponseMessage`, it's also possible the response message was not even created.